homeassistant/components/shelly/climate.py

```diff
@@ -0,0 +1,190 @@
+"""Climate support for Shelly."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, Final, cast
+
+from aioshelly.block_device import Block
+import async_timeout
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.components.shelly import BlockDeviceWrapper
+from homeassistant.components.shelly.entity import ShellyBlockEntity
+from homeassistant.components.shelly.utils import get_device_entry_gen
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import (
+    AIOSHELLY_DEVICE_TIMEOUT_SEC,
+    BLOCK,
+    DATA_CONFIG_ENTRY,
+    DOMAIN,
+    SHTRV_01_TEMPERATURE_SETTINGS,
+)
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up climate device."""
+
+    if get_device_entry_gen(config_entry) == 2:
+        return
+
+    wrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id][BLOCK]
+    for block in wrapper.device.blocks:
+        if hasattr(block, "targetTemp"):
+            async_add_entities([ShellyClimate(wrapper, block)])
+            break
+
+
+class ShellyClimate(ShellyBlockEntity, RestoreEntity, ClimateEntity):
+    """Representation of a KNX climate device."""
+
+    _attr_hvac_modes = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_AUTO]
+    _attr_icon = "hass:thermostat"
+    _attr_max_temp = SHTRV_01_TEMPERATURE_SETTINGS["max"]
+    _attr_min_temp = SHTRV_01_TEMPERATURE_SETTINGS["min"]
+    _attr_supported_features: int = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+    _attr_target_temperature_step = SHTRV_01_TEMPERATURE_SETTINGS["step"]
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, wrapper: BlockDeviceWrapper, block: Block) -> None:
+        """Initialize climate."""
+        super().__init__(wrapper, block)
+
+        self.wrapper = wrapper
+        self.block = block
+        self.control_result: dict[str, Any] | None = None
+
+        self._attr_name = self.wrapper.name
+        self._attr_unique_id = f"{self.wrapper.mac}-climate"
+        self._attr_preset_modes: list[str] = wrapper.device.settings[
+            "schedule_profile_names"
+        ]
+        self._attr_preset_modes.insert(0, PRESET_NONE)
```

Comment:
You're now changing the original list of Shelly. Don't do that, make a copy instead:
```suggestion
        self._attr_preset_modes: list[str] = [
            PRESET_NONE,
            *wrapper.device.settings["schedule_profile_names"]
        ]
```